src/SIUnits.jl

```diff
@@ -7,15 +7,28 @@ module SIUnits
     import Base: ==, +, -, *, /, .+, .-, .*, ./, //, ^
     import Base: promote_rule, promote_type, convert, show, mod
 
-    immutable SIQuantity{T<:Number,m,kg,s,A,K,mol,cd,rad,sr} <: Number
+    typealias UnitTuple NTuple{9,Int64}
```

Comment:
I haven't tried it with rational exponents, but there doesn't seem to be
anything that prevents it. Does it currently work properly with rational
exponents?

On Thu, Nov 19, 2015 at 11:22 PM Tomas Lycken <notifications@github.com>
wrote:

> In src/SIUnits.jl
> <https://github.com/Keno/SIUnits.jl/pull/78#discussion_r45441519>:
>
> > @@ -7,15 +7,28 @@ module SIUnits
> >      import Base: ==, +, -, *, /, .+, .-, .*, ./, //, ^
> >      import Base: promote_rule, promote_type, convert, show, mod
> >
> > -    immutable SIQuantity{T<:Number,m,kg,s,A,K,mol,cd,rad,sr} <: Number
> > +    typealias UnitTuple NTuple{9,Int64}
>
> Is Int64 really the correct thing here? Will this still be able to
> represent units like s^(1/2)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Keno/SIUnits.jl/pull/78/files#r45441519>.
>